pkg/streamingpromql/aggregations/common.go

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package aggregations
+
+import (
+	"github.com/prometheus/prometheus/promql/parser"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/functions"
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+// AggregationFunction accumulates series that have been grouped together and computes the output series data.
+type AggregationFunction interface {
```

Comment:
Perhaps a name like `AggregationGroup` would be clearer? This interface doesn't represent an aggregation function, but a single output group from an aggregation.

Reply:
I don't really like either naming. It's a method that operates on a group. However it seems like Prometheus leans away from calling aggregations functions so I'll go with AggregationGroup for now.